src/Error/ExceptionTrap.php

```diff
@@ -363,7 +366,6 @@ public function logException(Throwable $exception, ?ServerRequestInterface $requ
                 $this->logger()->log($exception, $request);
             }
         }
-        $this->dispatchEvent('Exception.beforeRender', ['exception' => $exception]);
```

Comment:
I think this was put here so that this event is also fired from the ErrorHandlerMiddleware. Because we needed to bridge both the ErrorHandler and ErrorRenderer interfaces the middleware operates ExceptionTrap at a lower level.

Right now I think this event will be missing in the ErrorHandlerMiddleware which is where most application exceptions are handled in.

Reply:
If the `ErrorHandlerMiddleware` is configured to used the deprecated `ErrorHandler` then `ExceptionTrap` doesn't seem to be used at all, so I don't think the `Exception.beforeRender` event is fired in that case.

https://github.com/cakephp/cakephp/blob/630bd2d9749320e05d1ecadbdfa51d84d12a36e5/src/Error/Middleware/ErrorHandlerMiddleware.php#L154-L157

That said this change would cause the event to be not fired when an exception is caught by `ErrorHandlerMiddleware` even when `ExceptionTrap` is used. So I'll have to make changes to address that.